solr/core/src/java/org/apache/solr/handler/designer/SchemaDesignerAPI.java

```diff
@@ -1119,8 +1120,9 @@ protected Map<String, Object> buildResponse(
     Map<String, Object> response = new HashMap<>();
 
     DocCollection coll = zkStateReader().getCollection(mutableId);
-    if (coll.getActiveSlicesArr().length > 0) {
-      String coreName = coll.getActiveSlicesArr()[0].getLeader().getCoreName();
+    if (!coll.getActiveSlices().isEmpty()) {
+      List<Slice> activeSlices = new ArrayList<>(coll.getActiveSlices());
+      String coreName = activeSlices.getFirst().getLeader().getCoreName();
```

Comment:
I like the `.getFirst` versus teh magic array logic!  More expressive of what you are doing.

Reply:
why create an ArrayList wrapper?
And can we call getActiveSlices once to avoid potential concerns of what work it may do?